app/src/main/java/org/gdg/frisbee/android/onboarding/AppInviteLinkGenerator.java

```diff
@@ -63,4 +78,30 @@ private String createDeepLink(String gplusId) {
             .toString();
     }
 
+    private void shareShortAppInviteLink(String longUrl) {
+        FirebaseDynamicLinksHub firebaseDynamicLinksHub = App.from(activity).getFirebaseDynamicLinksHub();
+        String requestString = "{\"longDynamicLink\": \"" + longUrl + "\",\"suffix\": {\"option\": \"SHORT\"}}";
+        RequestBody body = RequestBody.create(okhttp3.MediaType.parse("application/json; charset=utf-8"), requestString);
+        firebaseDynamicLinksHub.getShortenedUrl(BuildConfig.FIREBASE_WEB_API_KEY, body).enqueue(new Callback<FirebaseDynamicLinksResponse>() {
+            @Override
+            public void onSuccess(FirebaseDynamicLinksResponse response) {
+                createChooser(HttpUrl.parse(response.getShortLink()));
+                shareProgressDialog.dismiss();
```

Comment:
Let's try to decouple the UI concerns and the actual link shortening. 

Now the link creation is asynchronous. Ideally we should have a Listener with a possible method `onDynamicLinkReady(String url)`. Listener's method would be called with the shortened link when it is successful. 

Then the place where we implement this Listener can handle progress show/dismiss. 

What do you think?